test/cuda/tensors.jl

```diff
@@ -0,0 +1,648 @@
+                CUDA.@allowscalar begin
+                    @test norm(t + t, p) ≈ 2 * norm(t, p)
+                end
+                @test norm(t) ≈ norm(t')
+
+                t2 = @constinferred rand!(similar(t))
+                β = rand(T)
+                #@test @constinferred(dot(β * t2, α * t)) ≈ conj(β) * α * conj(dot(t, t2)) # broken for Irrep[CU₁]
+                @test dot(β * t2, α * t) ≈ conj(β) * α * conj(dot(t, t2))
+                @test dot(t2, t) ≈ conj(dot(t, t2))
+                @test dot(t2, t) ≈ conj(dot(t2', t'))
+                @test dot(t2, t) ≈ dot(t', t2')
+
+                i1 = @constinferred(isomorphism(T, V1 ⊗ V2, V2 ⊗ V1))
+                i2 = @constinferred(isomorphism(CuVector{T}, V2 ⊗ V1, V1 ⊗ V2))
+                CUDA.@allowscalar begin
+                    #@test i1 * i2 == @constinferred(id(T, V1 ⊗ V2)) # TODO
+                    #@test i2 * i1 == @constinferred(id(CuVector{T}, V2 ⊗ V1)) # TODO
+                    w = @constinferred(isometry(T, V1 ⊗ (oneunit(V1) ⊕ oneunit(V1)), V1))
+                    @test dim(w) == 2 * dim(V1 ← V1)
+                    @test w' * w == id(CuVector{T}, V1)
+                    @test w * w' == (w * w')^2
+                end
+            end
+        end=#
+        #=@timedtestset "Trivial space insertion and removal" begin
+            W = V1 ⊗ V2 ⊗ V3 ← V4 ⊗ V5
+            for T in (Float32, ComplexF64)
+                t = @constinferred CUDA.rand(T, W)
+                t2 = @constinferred insertleftunit(t)
+                @test t2 == @constinferred insertrightunit(t)
+                @test numind(t2) == numind(t) + 1
+                @test space(t2) == insertleftunit(space(t))
+                @test scalartype(t2) === T
+                @test t.data === t2.data
+                @test @constinferred(removeunit(t2, $(numind(t2)))) == t
+                t3 = @constinferred insertleftunit(t; copy = true)
+                @test t3 == @constinferred insertrightunit(t; copy = true)
+                @test t.data !== t3.data
+                for (c, b) in blocks(t)
+                    @test b == block(t3, c)
+                end
+                @test @constinferred(removeunit(t3, $(numind(t3)))) == t
+                t4 = @constinferred insertrightunit(t, 3; dual = true)
+                @test numin(t4) == numin(t) && numout(t4) == numout(t) + 1
+                for (c, b) in blocks(t)
+                    @test b == block(t4, c)
+                end
+                @test @constinferred(removeunit(t4, 4)) == t
+                t5 = @constinferred insertleftunit(t, 4; dual = true)
+                @test numin(t5) == numin(t) + 1 && numout(t5) == numout(t)
+                for (c, b) in blocks(t)
+                    @test b == block(t5, c)
+                end
+                @test @constinferred(removeunit(t5, 4)) == t
+            end
+        end=#
+        if hasfusiontensor(I)
+            @timedtestset "Basic linear algebra: test via conversion" begin
```

Comment:
How do you feel about `"test via CPU"` instead? I.e. instead of checking that the linear algebra commutes with `convert(Array, x)`, we check if it commutes with `collect`. This means that we can still test methods that don't have `hasfusiontensor`, and this should be faster, while being as good of a test.
Additionally, this avoids the issue about how to deal with array conversions from GPU for now

Reply:
`collect` should be ok for now